### PR TITLE
tests(etcdserver): add `server_access_control_test.go`

### PR DESCRIPTION
**Pull request by wafuwafu13**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/etcdserver/server_access_control_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+	for i, tt := range tests {
+		allowed := tt.accessController.OriginAllowed(tt.origin)
+		if allowed != tt.allowed {
+			t.Fatalf("#%d: allowd = %t, want %t", i, allowed, tt.allowed)
+		}
+	}
+}
+
+func TestIsHostWhitelisted(t *testing.T) {
```
Comment (ptabor) [inline server/etcdserver/server_access_control_test.go:70]: We should eventually change the name to 'allowlisted'. 
But it's outside of this PR.

**Review by ptabor — APPROVED** [review]
Thank you for providing the test.

File: server/etcdserver/server_access_control_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
+// Copyright 2016 The etcd Authors
```
Comment (ahrtr) [inline server/etcdserver/server_access_control_test.go:1]:
```suggestion
// Copyright 2022 The etcd Authors
```

File: server/etcdserver/server_access_control_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+			"https://example.com",
+			true,
+		},
+	}
+
+	for i, tt := range tests {
+		allowed := tt.accessController.OriginAllowed(tt.origin)
+		if allowed != tt.allowed {
+			t.Fatalf("#%d: allowd = %t, want %t", i, allowed, tt.allowed)
+		}
```
Comment (ahrtr) [inline server/etcdserver/server_access_control_test.go:66]: Minor comment: Please consider to use "github.com/stretchr/testify/assert"

File: server/etcdserver/server_access_control_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+			"example.com",
+			true,
+		},
+	}
+
+	for i, tt := range tests {
+		whitelisted := tt.accessController.IsHostWhitelisted(tt.host)
+		if whitelisted != tt.whitelisted {
+			t.Fatalf("#%d:whitelisted = %t, want %t", i, whitelisted, tt.whitelisted)
+		}
```
Comment (ahrtr) [inline server/etcdserver/server_access_control_test.go:117]: same as above

**Comment by wafuwafu13** [comment]
@ahrtr 
Thank you for reviewing. I fixed all.

**Pull request merged**